singleheader/CMakeLists.txt

```diff
@@ -67,32 +97,34 @@ else(MSVC)
       DEPENDS ${SINGLEHEADER_FILES}
     )
   endif()
-
   #
   # "make amalgamate" to generate the header files directly and update the original source
   #
   add_custom_target(amalgamate DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/simdjson.cpp ${CMAKE_CURRENT_SOURCE_DIR}/simdjson.h ${CMAKE_CURRENT_SOURCE_DIR}/amalgamate_demo.cpp ${CMAKE_CURRENT_SOURCE_DIR}/README.md)
-  
+  add_dependencies(simdjson amalgamate)
 endif(MSVC)
 
-
 #
-# Include this if you intend to #include "simdjson.cpp" in your own .cpp files.
+# Installation
 #
-add_library(simdjson-singleheader-include-source INTERFACE)
-target_include_directories(simdjson-singleheader-include-source INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
+install(FILES simdjson.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

Comment:
I feel like we need to install simdjson.cpp somewhere too; we do expose that as a public interface (you don't have to use the library). However, I wouldn't do that in this PR, let's sketch it out separate.

Reply:
I assumed since we are installing libsimdjson.so I didn't install the source file and that is also why I removed #include simdjson.cpp from amalgamate_demo, but if it is part of the public API I can open a new PR after this one